src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -72,9 +78,68 @@ def _set_sanitizer_options(fuzzer_path):
   environment.set_memory_tool_options(sanitizer_options_var, sanitizer_options)
 
 
+def _parse_centipede_stats(
+    stats_file: str) -> Optional[Dict[str, Union[int, float]]]:
+  """Parses the Centipede stats file and returns a dictionary with labels
+  and their respective values.
+
+  Args:
+      stats_file: the path to Centipede stats file.
+
+  Returns:
+      a dictionary containing the stats.
+  """
+  if not os.path.exists(stats_file):
+    return None
+  with open(stats_file, 'r') as statsfile:
+    csvreader = csv.reader(statsfile)
+    l = list(csvreader)
+    # If the binary could not run at all, the file will be empty or with only
+    # the column description line.
+    if len(l) <= 1:
+      return None
+    return {
+        l[0][i]: float(l[-1][i]) if '.' in l[-1][i] else int(l[-1][i])
```

Comment:
Can you add a comment to explain what this is doing? Or to include an example line that we're parsing? 

Also -- is "." reliable enough of an indicator that something is a float? e.g. is "1e-1" a possible input here? 

It'll likely be a little more robust if we just use `isdigit` to check for int validity instead.

Reply:
Arg.. I meant to rework the function but actually forgot to do it. WDYT now?